src/core/views.py

```diff
@@ -2775,3 +2780,153 @@ def post(self, request, *args, **kwargs):
+    pk_url_kwarg = 'organization_name_id'
+    success_url = reverse_lazy('core_edit_profile')
+    success_message = _("Custom organization updated: %(value)s")
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['account'] = self.request.user
+        return context
+
+
+@method_decorator(login_required, name='dispatch')
+class AffiliationCreateView(SuccessMessageMixin, CreateView):
+    """
+    Allows a user to create a new affiliation for themselves.
+    """
+
+    model = core_models.Affiliation
+    form_class = forms.AffiliationForm
+    success_url = reverse_lazy('core_edit_profile')
+    success_message = _("Affiliation created: %(organization)s")
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['account'] = self.request.user
+        organization_id = self.kwargs.get('organization_id')
+        context['organization'] = core_models.Organization.objects.get(
+            pk=organization_id
+        )
+        return context
+
+    def get_initial(self):
+        initial = super().get_initial()
+        initial['account'] = self.request.user
+        initial['organization'] = self.kwargs.get('organization_id')
+        return initial
+
+
+@method_decorator(login_required, name='dispatch')
+class AffiliationUpdateView(SuccessMessageMixin, UpdateView):
+    """
+    Allows a user to update one of their own affiliations.
+    """
+
+    model = core_models.Affiliation
+    pk_url_kwarg = 'affiliation_id'
+    form_class = forms.AffiliationForm
+    success_url = reverse_lazy('core_edit_profile')
+    success_message = _("Affiliation updated: %(organization)s")
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['account'] = self.request.user
+        affiliation_id = self.kwargs.get('affiliation_id')
+        context['affiliation'] = core_models.Affiliation.objects.get(
+            pk=affiliation_id
+        )
+        context['organization'] = context['affiliation'].organization
+        return context
+
+
+@method_decorator(login_required, name='dispatch')
+class AffiliationDeleteView(SuccessMessageMixin, DeleteView):
+    """
+    Allows a user to delete one of their own affiliations.
+    """
+
+    model = core_models.Affiliation
+    pk_url_kwarg = 'affiliation_id'
+    success_url = reverse_lazy('core_edit_profile')
+    success_message = _("Affiliation deleted")
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['account'] = self.request.user
+        affiliation_id = self.kwargs.get('affiliation_id')
+        context['affiliation'] = core_models.Affiliation.objects.get(
+            pk=affiliation_id
+        )
+        context['organization'] = context['affiliation'].organization
+        return context
```

Comment:
Have we discussed introducing the use of Django's generic CBVs?
They don't seem to provide any value in terms of reusability, have added cognitive load over existing views and more importantly, they are inconsistent with the rest of the codebase

Reply:
> Have we discussed introducing the use of Django's generic CBVs? They don't seem to provide any value in terms of reusability, have added cognitive load over existing views and more importantly, they are inconsistent with the rest of the codebase

Not sure I agree that "they don't seem to provide any value in terms of reusability". Let's schedule a time to talk about your concerns.